.github/workflows/rust.yml

```diff
@@ -0,0 +1,25 @@
+name: "Rust Format, Clippy and Test"
+
+on:
+  push:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Format
+      run: make format-check
+    - name: Clippy
+      run: make clippy
```

Comment:
For local development this is fine, but in the CI we rather prefer to mark the warnings as errors, something like `-- -D warning` should do the job.